blivet/tasks/fsresize.py

```diff
@@ -115,6 +121,54 @@ def args(self):
         ]
 
 
+class XFSResize(FSResize):
+    ext = availability.XFSRESIZE_APP
+    unit = B
+    size_fmt = None
+
+    @contextmanager
+    def _do_temp_mount(self):
+        if self.fs.status:
+            yield
+        else:
+            dev_name = os.path.basename(self.fs.device)
+            tmpdir = tempfile.mkdtemp(prefix="xfs-tempmount-%s" % dev_name)
+            log.debug("mounting XFS on '%s' to '%s' for resize", self.fs.device, tmpdir)
+            try:
+                self.fs.mount(mountpoint=tmpdir)
+            except FSError as e:
+                raise FSError("Failed to mount XFS filesystem for resize: %s" % str(e))
+
+            try:
+                yield
+            finally:
+                util.umount(mountpoint=tmpdir)
+                os.rmdir(tmpdir)
+
+    def _get_block_size(self):
+        if self.fs._current_info:
+            # this should be set by update_size_info()
+            for line in self.fs._current_info.split("\n"):
+                if line.startswith("blocksize ="):
+                    return int(line.split("=")[-1])
+
+        raise FSError("Failed to get XFS filesystem block size for resize")
+
+    def size_spec(self):
+        # size for xfs_growfs is in blocks
+        return str(self.fs.target_size.convert_to(self.unit) / self._get_block_size())
+
+    @property
+    def args(self):
+        return [self.fs.system_mountpoint, "-D", self.size_spec()]
+
+    def do_task(self):
+        """ Resizes the XFS format. """
+
+        with self._do_temp_mount():
```

Comment:
XFS must be mounted for resize. I've decided to mount the device if it isn't mounted, but it might be safer to let user decide and mount the device manually. We could simply add check to `_pre_resize`. Thoughts?

Reply:
The way you have it now seems reasonable to me.